test/common/network/happy_eyeballs_connection_impl_test.cc

```diff
@@ -172,9 +172,9 @@ TEST_F(HappyEyeballsConnectionImplTest, DisallowedFunctions) {
   startConnect();
 
   EXPECT_ENVOY_BUG(connection_callbacks_[0]->onAboveWriteBufferHighWatermark(),
```

Comment:
In addition to these happy eyeballs tests, we should add new unit tests for the new MultiConnectionBaseImpl class itself.

Reply:
I've split happy_eyeballs_connection_impl_test.cc` into `multi_connection_base_impl_test.cc` and `happy_eyeballs_connection_provider_test.cc`, IMO most of the test cases should be moved to `multi_connection_base_impl_test`